### PR TITLE
feat(sort): warn when an explicit memory budget crowds the host

### PR DESCRIPTION
**Pull request by nh13**
The memory-budget resolver already warns when the budget exceeds total host RAM. The more common pipeline footgun is a budget that merely *approaches* it: with an aligner feeding the sort and holding a ~10-15 GiB genome index resident, an explicit `--max-memory` that fits on its own can still OOM the host. (Concretely: a 32 GB host was OOM-killed by a `--max-memory 2GiB × 8 = 16 GB` sort buffer co-resident with a ~15 GB aligner index.)

## Change

- **`resolve_memory_budget_with_total` (common.rs)** — after the existing `budget > total` warning, add an `else if budget > 80% × total` branch that warns the budget leaves little headroom for co-resident processes and names the likely culprit (an aligner holding a genome index). Shared by every pipeline command via `QueueMemoryOptions`.
- **No behavior change to the budget.** The resolver cannot see the co-resident aligner and cannot know how much headroom it needs, so it warns and leaves the explicit number intact — it does not silently shrink it, and `--memory-reserve` stays `auto`-only.
- **Docs** — `--max-memory`s help now states it bounds the in-memory record buffer, not total RSS (ingest decompression/parse buffers and the final merge per-file read-ahead sit on top), so peak RSS runs somewhat higher; size pipe headroom accordingly.

## Tests

`test_fixed_budget_high_fraction_of_host_warns_but_is_unchanged`: a 14 GiB budget on a 16 GiB host (87.5%) exercises the new branch and asserts the returned budget is unshrunk. `cargo test -p fgumi` green; clippy and fmt clean.

Risk: command output changes: none; `unsafe` changes: none, and `CLAUDE.md` allowlist changes: none; memory bound policy: none—the explicit memory budget remains unchanged, and the change only adds a warning.

- Warn when an explicit memory budget exceeds 80% of host RAM but remains within total RAM.
- Keep automatic budgets and over-total budgets on their existing paths.
- Clarify that `--max-memory` limits the in-memory record buffer, while read-ahead buffers add to peak RSS.
- Add tests for threshold boundaries, budget-type exclusions, warning emission, and budget preservation.

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai pause

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

File: src/lib/commands/common.rs [hunk 1]
```diff
@@ -2017,6 +2062,41 @@ mod tests {
...
+
+        fn flush(&self) {}
+    }
+
+    /// Install [`CaptureLogger`] as the global logger so subsequent emitted
+    /// records accumulate in [`CAPTURED_LOGS`]. Call once per test process
+    /// (nextest isolates each test), in place of `enable_logging`.
+    fn capture_logs() {
+        log::set_logger(&CAPTURE_LOGGER).expect("a logger was already installed for this test");
+        log::set_max_level(log::LevelFilter::Trace);
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/lib/commands/common.rs]: Fixed in c4cc2a60. `enable_logging` and `capture_logs` now both route through a single `install_capture_logger()` that installs the inspectable `CaptureLogger` exactly once per process (guarded by `std::sync::Once`), so there is no second `log::set_logger` call to panic under a shared-process `cargo test` run (nextest still isolates per process). `capture_logs` clears `CAPTURED_LOGS` before the operation so the assertion sees only what its own call emits. `env_logger` is no longer installed in tests (it remains the production logger in `main.rs`).

**Comment by nh13** [comment]
@coderabbitai review
